lib/features/home/data/repository/jobs_repository_impl.dart

```diff
@@ -0,0 +1,135 @@
+import 'package:dartz/dartz.dart';
+import 'package:final_project/core/const/strings.dart';
+import 'package:final_project/core/error/failure.dart';
+import 'package:final_project/features/home/data/models/apply/apply_sent_data.dart';
+import 'package:final_project/features/home/data/models/jobs/jobs_model.dart';
+import 'package:final_project/features/home/data/services/remote/jobs_service.dart';
+import 'package:final_project/features/home/domain/entities/job.dart';
+import 'package:final_project/features/home/domain/repository/jobs_repository.dart';
+import 'package:flutter/foundation.dart';
+import 'package:injectable/injectable.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+@Injectable(as: JobsRepository)
+class JobsRepositoryImpl extends JobsRepository {
+  final JobsService service;
+
+  JobsRepositoryImpl(this.service);
+
+  @override
+  Future<Either<Failure, List<Job>>> getJobs() async {
+    final _token = await getToken();
+    try {
+      if (kDebugMode) {
+        print('Get Jobs start...');
+      }
+      if (_token != null && _token.isNotEmpty) {
+        final jobsModel = await service.getJobs(token: 'Bearer $_token');
+        if (kDebugMode) {
+          print('Get Job user model $jobsModel');
+        }
+        return right(_jobMapper(jobsModel));
+      } else {
+        return left(const Failure('No Token'));
```

Comment:
Users should never be shown errors that contain technical terminologies. Try generic error messages or an understanble ones like "Something went wrong"